pkg/cmd/experimental/syncgroups/schema1/newsyncer.go

```diff
@@ -0,0 +1,58 @@
+package schema1
+
+import (
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups"
+	syncconfig "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/config"
+)
+
+// NewGroupSyncer creates a GroupSyncer for a schema-specific config written for an LDAP schema with
+// first-class group entries that hold membership lists as attributes
+func NewGroupSyncer(ldapSyncConfig LDAPSyncConfig) (syncgroups.GroupSyncer, error) {
```

Comment:
Move this into a stubbed out command package.  Flag it with a TODO to be deleted.  Have it accept the absolute minimum arguments you need to construct a job that runs.  Default absolutely everything you can.  It should pretty much be `ldapurl string`.

That will help you start to factor this more effectively.  You are still conflating user config about making choice with objects used to configure the runtime bits.